rust/src/storage/s3/dynamodb.rs

```diff
@@ -0,0 +1,427 @@
+use std::collections::HashMap;
```

Comment:
This file should probably be named like dynamodblock or something like that. Once it gets into a stable state, we should release it as a standalone crate :)

Reply:
wdyt of placing this file, should it be under s3 module, or under storage itself?

Reply:
i think s3 would be better since it's only needed for s3 for now. when we release it as a standalone crate, we can relocate it to the repo root or create a new github repo for it.

Reply:
yeap, makes sense